### PR TITLE
search added

### LINKED ISSUE #5: Add Search option.

### PR DESCRIPTION
**Pull request by nuclrya**
Fixes #5 
-
i have done most of the part but there are some bugs left which i m not able to solve
like repo number is initialized to one when searching for new term

### CODE REVIEW

**Comment by arshadkazmi42** [comment]
There are couple of things.
1. This is the URL you should use
`https://api.github.com/search/issues?page=${pageNo}&per_page=${perPage}&q=${this.state.searchTerm}&author:hacktoberfest-team`

2. Do not have separate function for search, let's use `callPage` function itself, as that is handling all the logic, you can just modify the url if the search term is present.

I think that should fix all the issues.
Let me know if you need any other information

> i have done most of the part but there are some bugs left which i m not able to solve
> like repo number is initialized to one when searching for new term

**Comment by NuCelyA** [comment]
@arshadkazmi42 i have updated code 
plz see this if this is fine

File: src/App.js [hunk 1]
```diff
@@ -26,14 +28,22 @@ class App extends Component{
       ...this.state,
       loader:true
     })
-    fetch(`https://api.github.com/search/issues?page=${pageNo}&per_page=${perPage}&q=author:hacktoberfest-team`)
+    let searchUrl = ""
+    if(this.state.searchTerm.length === 0){
+      searchUrl = `https://api.github.com/search/issues?page=${pageNo}&per_page=${perPage}&q=author:hacktoberfest-team`
+    }
+    else{
+      searchUrl = `https://api.github.com/search/issues?page=${pageNo}&per_page=${perPage}&q=${this.state.searchTerm}&author:hacktoberfest-team`
```
Comment (arshadkazmi42) [inline src/App.js]: Hey, so I was checking this URL properly and it seems this is not searching for issues we need.
I was thinking of making a local search, which will filter already loaded data.
As API also has limit, so calling it multiple times,  it blocks it for few minutes.

Let's build a local search. Any thoughts?

Comment (NuCelyA) [reply src/App.js]: well, i m just a beginner in react and know little about it but if u could guide me how to do task, i could give it a try
and for api limit , i think there is some github api tokens through which we can increase its limit

Comment (arshadkazmi42) [reply src/App.js]: > and for api limit , i think there is some github api tokens through which we can increase its limit

We do not want to take this up right now, as we have a lot other other things to be build around the app.

> well, i m just a beginner in react and know little about it but if u could guide me how to do task, i could give it a try

For this, let's wait for #17  to get merged. 
Once that is merged, you could add a filtering function at here https://github.com/akramkazmi71/hacktoberfest-xrepos/blob/b5a21915675211de572eea1121b47df7eaf210ef/src/App.js#L81

Apply the filtering on `repName`.

Comment (arshadkazmi42) [reply src/App.js]: @NuCelyA All changes are merged, you can implement your search options before this line

https://github.com/akramkazmi71/hacktoberfest-xrepos/blob/master/src/App.js#L45

**Comment by arshadkazmi42** [comment]
@NuCelyA Any updates on this?

**Comment by arshadkazmi42** [comment]
Inactive PR

**Pull request closed without merging**